runtime/vm/JFRChunkWriter.cpp

```diff
@@ -1273,4 +1273,38 @@ VM_JFRChunkWriter::writeThreadDumpEvent()
 
 	return dataStart;
 }
+
+void
+VM_JFRChunkWriter::writeSystemProcessEvent(void *anElement, void *userData)
+{
+	SystemProcessEntry *entry = (SystemProcessEntry *)anElement;
+	VM_JFRChunkWriter *_writer = (VM_JFRChunkWriter *)userData;
+	VM_BufferWriter *_bufferWriter = _writer->_bufferWriter;
+	PORT_ACCESS_FROM_JAVAVM(_writer->_vm);
+
+	/* Reserve size field */
+	U_8 *dataStart = _writer->reserveEventSize(_bufferWriter);
+
+	/* Write event type */
+	_bufferWriter->writeLEB128(SystemProcessID);
+
+	/* Write start time */
+	_bufferWriter->writeLEB128(entry->ticks);
+
+	/* Write process ID as string */
+	_writer->writeStringLiteral(entry->pid);
```

Comment:
Use a local buffer to convert the process identifier to a string here (rather than allocate it elsewhere).